### PR TITLE
ADEN-9983 Don't go to UAP sticked if viewability+timeout is counted

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: platforms/ucp/templates/helpers/bfaa-helper.ts [hunk 1]
```diff
@@ -90,4 +92,18 @@ export class BfaaHelper {
 
 		return () => window.scrollBy(0, elementOfReference.getBoundingClientRect().top - startValue);
 	}
+
+	isViewedAndDelayed(): Observable<unknown> {
+		const slotViewed: Promise<void> = this.adSlot.loaded.then(() => this.adSlot.viewed);
+
+		return from(slotViewed.then(() => utils.wait(this.getAdditionalStickinessTime())));
```
Comment (jacekbj) [inline platforms/ucp/templates/helpers/bfaa-helper.ts]: Consider using [concat](https://www.learnrxjs.io/learn-rxjs/operators/combination/concat) and [timer](https://www.learnrxjs.io/learn-rxjs/operators/creation/timer):
```
return concat(
 from(this.adSlot.loaded),
 from(this.adSlot.viewed),
 timer(this.getAdditionalStickinessTime()),
);
```

File: platforms/ucp/templates/handlers/bfaa-impact-handler.ts [hunk 2]
```diff
@@ -30,7 +30,19 @@ export class BfaaImpactHandler implements TemplateStateHandler {
 		this.helper = new BfaaHelper(this.manipulator, this.params, this.adSlot, navbar);
 	}
 
-	async onEnter(transition: TemplateTransition<'sticky'>): Promise<void> {
+	async onEnter(transition: TemplateTransition<'sticky' | 'resolved'>): Promise<void> {
+		let shouldStick = true;
```
Comment (jacekbj) [inline platforms/ucp/templates/handlers/bfaa-impact-handler.ts]: It's possible to remove the local variable `shouldStick`:
```
const isViewedAndDelayed$ = this.helper.isViewedAndDelayed().pipe(
  takeUntil(this.unsubscribe$),
  shareReplay(1),
);
```
and later.... (see next comment)

File: platforms/ucp/templates/handlers/bfaa-impact-handler.ts [hunk 3]
```diff
@@ -58,7 +70,11 @@ export class BfaaImpactHandler implements TemplateStateHandler {
 				tap(() => {
 					const correction = this.helper.usePositionCorrection(this.footer);
 
-					transition('sticky').then(correction);
+					if (shouldStick) {
```
Comment (jacekbj) [inline platforms/ucp/templates/handlers/bfaa-impact-handler.ts]: ...and:
```
filter(() => this.reachedResolvedSize()),
withLatestFrom(isViewedAndDelayed$),
tap(([_, shouldStick]) => {
  const correction = this.helper.usePositionCorrection(this.footer);
  if (shouldStick) {
    transition('sticky').then(correction);
  } else {
    transition('resolved').then(correction);
  }
})
```

File: platforms/ucp/templates/helpers/bfaa-helper.ts [hunk 4]
```diff
@@ -90,4 +92,18 @@ export class BfaaHelper {
...
+	}
+
+	private getAdditionalStickinessTime(): number {
+		if (!isUndefined(this.params.stickyAdditionalTime)) {
+			return this.params.stickyAdditionalTime;
```
Comment (jacekbj) [inline platforms/ucp/templates/helpers/bfaa-helper.ts]: Can't wait for TS upgrade! `return return this.params?.stickyAdditionalTime ?? universalAdPackage.BFAA_UNSTICK_DELAY`

Comment (dmnsobczak) [reply platforms/ucp/templates/helpers/bfaa-helper.ts]: Yeah. I was joking about that at the beginning but this is very useful change :)

File: platforms/ucp/templates/helpers/bfaa-helper.ts (hunk 1 above)
Comment (dmnsobczak) [reply platforms/ucp/templates/helpers/bfaa-helper.ts]: I actually moved this method without thinking about improvements. Thanks for the input!
...but it's not working. The tap is executed three times after every element of the concat :/

File: platforms/ucp/templates/handlers/bfaa-impact-handler.ts (hunk 2 above)
Comment (dmnsobczak) [reply platforms/ucp/templates/handlers/bfaa-impact-handler.ts]: Nice idea! Changed.

File: platforms/ucp/templates/handlers/bfaa-impact-handler.ts (hunk 3 above)
Comment (dmnsobczak) [reply platforms/ucp/templates/handlers/bfaa-impact-handler.ts]: 👍

File: platforms/ucp/templates/helpers/bfaa-helper.ts (hunk 1 above)
Comment (jacekbj) [reply platforms/ucp/templates/helpers/bfaa-helper.ts]: Lets just leave it as is. I don't know if there is a better solution than 3x`switchMap`

**Pull request merged**